packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,162 @@
+    this._cache = new Cache({
+      getCb: (address) => this.getAccountFromProvider(address),
+      putCb: (_address, _account) => Promise.resolve(),
+      deleteCb: (_address) => Promise.resolve(),
+    })
+  }
+
+  async putContractCode(address: Address, value: Buffer): Promise<void> {
+    // Store contract code in the cache
+    this.contractCache.set(address.toString(), value)
+  }
+
+  async getContractStorage(address: Address, key: Buffer): Promise<Buffer> {
+    const slotCacheKey = `${address.toString()}--0x${key.toString('hex')}`
+    // Check storage slot in cache
+    let storage: Buffer | string | undefined = this.storageCache.get(slotCacheKey)
+    if (typeof storage !== 'undefined') return storage
+    // Retrieve storage slot from provider if not found in cache
+    storage = await this.provider.getStorageAt(address.toString(), bufferToBigInt(key))
+    const value = toBuffer(storage)
+    // Cache retrieved storage slot
+    await this.putContractStorage(address, key, value)
+    return value
+  }
+
+  async putContractStorage(address: Address, key: Buffer, value: Buffer): Promise<void> {
+    // Set value in storageCache with `[address]--[hexKey]` formatted key
+    this.storageCache.set(`${address.toString()}--${bufferToHex(key)}`, value)
+  }
+
+  copy(): EthersStateManager {
+    return new EthersStateManager({ common: this._common.copy(), provider: this.provider })
+  }
+
+  dumpStorage(_address: Address): Promise<StorageDump> {
+    throw new Error('Method not implemented.')
+  }
+
+  async accountExists(address: Address): Promise<boolean> {
+    log(`seeing if ${address.toString()} exists`)
+    const account = this._cache.get(address)
+    if (account.isEmpty()) {
+      // Get latest block (or block specified in `this.blockTag`)
+      const block = await this.provider.send('eth_getBlockByNumber', [this.blockTag, false])
+      // Get merkle proof for `address` from provider
+      const proof = await this.provider.send('eth_getProof', [address.toString(), [], block.number])
+      const proofBuf = proof.accountProof.map((proofNode: string) => toBuffer(proofNode))
+      const trie = new SecureTrie()
+      const verified = await trie.verifyProof(toBuffer(block.stateRoot), address.buf, proofBuf)
+      // if not verified (i.e. verifyProof returns null), account does not exist
+      if (verified === null) return false
+    }
+    return true
+  }
+
+  async getAccount(address: Address): Promise<Account> {
+    const account = await this._cache.getOrLoad(address)
+    return account
+  }
+  async getAccountFromProvider(address: Address): Promise<Account> {
+    log(`getting account data for ${address.toString()}`)
+    let accountData
+    try {
+      accountData = await this.provider.send('eth_getProof', [
+        address.toString(),
+        [],
+        this.blockTag,
+      ])
+    } catch (e) {
+      accountData.balance = await this.provider.getBalance(address.toString(), this.blockTag)
```

Comment:
I think we should remove this catch block and throw entirely instead, since we have no way to get `storageRoot` for an account over RPC.